src/realm/node_header.hpp

```diff
@@ -612,41 +498,39 @@ class NodeHeader {
 
     static inline void set_flags(char* header, uint8_t flags)
     {
-        if (get_kind(header) == 'A') {
-            REALM_ASSERT(flags <= 7);
-            auto h = (uint8_t*)header;
-            h[4] = (h[4] & 0b00011111) | flags << 5;
-        }
-        else {
-            auto h = (uint8_t*)header;
-            h[2] = (h[2] & 0b00001111) | flags << 4;
-        }
+        REALM_ASSERT(flags <= 7);
+        auto h = (uint8_t*)header;
+        h[4] = (h[4] & 0b00011111) | flags << 5;
     }
     static inline uint8_t get_flags(char* header)
     {
-        if (get_kind(header) == 'A') {
-            auto h = (uint8_t*)header;
-            return h[4] >> 5;
-        }
-        else {
-            auto h = (uint8_t*)header;
-            return h[2] >> 4;
-        }
+        auto h = (uint8_t*)header;
+        return h[4] >> 5;
+    }
+
+    static inline void set_flags2(char* header, uint8_t flags)
```

Comment:
Maybe, some better naming instead of flag1 and flag2...

Reply:
We'll cross that bridge when needed